api/v1alpha1/apidefinition_types.go

```diff
@@ -163,8 +170,34 @@ type URLRewriteMeta struct {
 	// MatchPattern is a regular expression pattern to match the path
 	MatchPattern string `json:"match_pattern"`
 	// RewriteTo is the target path on the upstream, or target URL we wish to rewrite to
-	RewriteTo string           `json:"rewrite_to"`
-	Triggers  []RoutingTrigger `json:"triggers"`
+	// +optionall
+	RewriteTo string `json:"rewrite_to,omitempty"`
+	// RewriteToInternal serves as rewrite_to but used when rewriting to target
+	// internal api's
+	// When rewrite_to and rewrite_to_loop are both provided then
+	// rewrite_to will take rewrite_to_loop
+	// +optionall
+	RewriteToInternal *LoopInternal `json:"rewrite_to_internal,omitempty"`
+	// +optionall
+	Triggers []RoutingTrigger `json:"triggers,omitempty"`
+}
+
+// LoopInternal defines options that constructs a url that refers to an api that
+// is loaded into the gateway.
+type LoopInternal struct {
+	// Self set this to true if the api definition is looping to itself.
+	Self bool `json:"self,omitempty"`
+	// API a namespaced/name to the api definition resource that you are
+	// targetting. If self is set to true this value will be ignored.
+	//	example
+	API string `json:"api,omitempty"`
```

Comment:
As per ticket update, check whether we should be able to route to an internal API in another namespace?

How about referencing the API Definition as:

```
target:
  namespace: foo
  name: bar
```

If the namespace is omitted - then reference the namespace of current object.

And if we don't need the namespace because we can only link to resources in the current namespace, we could just define the target as `target: string` to reference the name of target.

Reply:
done 33def54